sdk/core/core/inc/az_http_header.h

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_HTTP_HEADER_H
+#define AZ_HTTP_HEADER_H
+
+#include <az_span_emitter.h>
+#include <az_pair.h>
+
+#include <_az_cfg_prefix.h>
+
+/**
+ * Emits an HTTP header as a sequence of spans in a format "%{key}: %{value}"
+ */
+AZ_NODISCARD az_result
+az_http_header_emit_span_seq(az_pair const * const self, az_span_action const action);
+
+/**
+ * @az_http_header_emit_span_sequence as @az_span_emitter.
+ */
+AZ_ACTION_FUNC(az_http_header_emit_span_seq, az_pair const, az_span_emitter)
```

Comment:
again, I really prefer just writing out the typedef here. That way when I see az_http_header_emit_span_seq_action I can simply "go to definition" and see the function pointer definition.

Reply:
For this specific one, I don't have a safe solution without a macro. I may consider to remove `AZ_ACTION_TYPE` but it's almost impossible to do safe cast wtihout `AZ_ACTION_FUNC`.